### PR TITLE
Fix/linear transaction cost

### PR DESCRIPTION
**Pull request by edlins**
Fixes #.

Changes proposed in this pull request:
- Fix calculation of linear transaction cost
- Fix unit test cases
- 

- [ ] added an entry to the unreleased section of `CHANGES.md`

### CODE REVIEW

**Comment by edlins** [comment]
I really don't like assertEquals() with some hardcopied mystery number.  If you don't show your work it's really hard to validate.

File: ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/LinearTransactionCostCriterionTest.java [hunk 1]
```diff
@@ -34,17 +34,48 @@ this software and associated documentation files (the "Software"), to deal in
...
+        double exitCost = amount * a + b;
+        amount = amount - exitCost;
+        double totalCost = entryCost + exitCost;
+        System.out.println(entryCost + " " + profit + " " + amount + " " + exitCost + " " + totalCost);
+        // good form
+        assertEquals(totalCost, transactionCost.calculate(series, tradingRecord), TATestsUtils.TA_OFFSET);
+
         tradingRecord.operate(2);
         tradingRecord.operate(3);
-        assertEquals(24.3473, transactionCost.calculate(series, tradingRecord), TATestsUtils.TA_OFFSET);
```
Comment (edlins) [inline ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/LinearTransactionCostCriterionTest.java:44]: not very useful, and wrong

File: ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/LinearTransactionCostCriterionTest.java [hunk 2]
```diff
@@ -34,17 +34,48 @@ this software and associated documentation files (the "Software"), to deal in
...
+        assertEquals(24.3759, transactionCost.calculate(series, tradingRecord), TATestsUtils.TA_OFFSET);
```
Comment (edlins) [inline ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/LinearTransactionCostCriterionTest.java:58]: not very useful, but correct

File: ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/LinearTransactionCostCriterionTest.java [hunk 3]
```diff
@@ -34,17 +34,48 @@ this software and associated documentation files (the "Software"), to deal in
...
+
+        entryCost = amount * a + b;
+        profit = series.getBar(3).getClosePrice().dividedBy(series.getBar(2).getClosePrice()).doubleValue();
+        amount = (amount - entryCost) * profit;
+        exitCost = amount * a + b;
+        amount = amount - exitCost;
+        totalCost += entryCost + exitCost;
+        assertEquals(totalCost, transactionCost.calculate(series, tradingRecord), TATestsUtils.TA_OFFSET);
```
Comment (edlins) [inline ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/LinearTransactionCostCriterionTest.java:66]: much better

**Comment by team172011** [comment]
> I really don't like assertEquals() with some hardcopied mystery number. If you don't show your work it's really hard to validate.

Not the best way to validate correctness, but still a good solution to satisfy that everything else works as before after one pushed...

Since #111 we also have the possibility of excel sheet validation

**Comment by edlins** [comment]
I agree only that it's a check to make sure the results didn't change, but says nothing about correctness. 
 If the original developer has the algorithm wrong, they might do it wrong in the method, the unit test, and Excel.  It's best if a different person writes the unit tests, and they should do so in a way that demonstrates the logic in a transparent manner.  And the unit tests should be written before the methods..  Regardless, there should be a very high standard for thorough and transparent unit tests in a technical analysis library.  In fact, I care much more about clear unit tests than I do about clear method code.  I happily offer this unit test for consideration as a template for analysis criteria unit tests.

**Comment by edlins** [comment]
Will rewrite with xls validation.

**Pull request closed without merging**